app/routes/press-releases/press-release/edit.js

```diff
@@ -0,0 +1,19 @@
+import Route from '@ember/routing/route';
+
+export default class PressReleasesPressReleaseEditRoute extends Route {
+  beforeModel(transition) {
+    if (transition.from) {
+      if (transition.from.name === 'press-releases.overview.published') {
+        this.transitionTo('press-releases.press-release.published')
+      }
```

Comment:
Deze logica moet aangepast worden: er moet op basis van het model, en niet op basis van de `fromRoute` beslist worden of het persbericht al gepubliceerd is. Nu kan men voor een gepubliceerd persbericht toch nog zelf de edit-route openen door rechtstreeks naar deze route te gaan.
Aangezien het model van de parent route gebruikt kan worden (via `this.modelFor`) kan de logica in de `beforeModel`-hook uitgevoerd worden.